coalib/bearlib/aspects/Formatting.py

```diff
@@ -0,0 +1,355 @@
+        # the max number of lines per file is usually 999.
+        """
+        example_language = 'python'
+        importance_reason = """
+        Too long programs (or files) are difficult to read, maintain and
+        understand.
+        """
+        fix_suggestions = """
+        Splitting files into modules, writing shorter methods and classes.
+        """
+    max_file_length = Taste[int](
+        'Maximum number of line for a file',
+        (999,), default=999)
+
+
+@Formatting.subaspect
+class Spacing:
+    """
+    All whitespace found between non-whitespace characters.
+    """
+    class docs:
+        example = """
+        # Here is an example of code with spacing issues including
+        # unnecessary blank lines and missing space around operators.
+
+
+
+        def func(   ):
+           return      37*-+2
+        """
+        example_language = 'python'
+        importance_reason = """
+        Useless spacing affects the readability and maintainability of a code.
+        """
+        fix_suggestions = """
+        Removing the trailing spaces and the meaningless blank lines.
+        """
+
+
+@Spacing.subaspect
+class TrailingSpace:
+    """
+    Unnecessary whitespace at end of a line.
+
+    Trailing space is all whitespace found after the last non-whitespace
+    character on the line until the newline. This includes tabs "\\\\t",
+    blank lines, blanks etc.
+    """
+    class docs:
+        example = """
+        def func( a ):
+              pass
+
+        """.replace('\n', '\t\n')
+        example_language = 'python'
+        importance_reason = """
+        Trailing spaces make code less readable and maintainable.
+        """
+        fix_suggestions = """
+        Removing the trailing spaces.
+        """
+    allow_trailing_spaces = Taste[bool](
+        'Determines whether or not trailing spaces should be allowed or not.',
+        (True, False), default=False)
+
+
+@Spacing.subaspect
+class BlankLine:
+    """
+    A line with zero characters.
+    """
+    class docs:
+        example = """
+        name = input('What is your name?')
+
+
+        print('Hi, {}'.format(name))
+        """
+        example_language = 'python'
+        importance_reason = """
+        Various programming styles use blank lines in different places.
+        The usage of blank lines affects the readability, maintainability and
+        length of a code i.e blank lines can either make code longer, less
+        readable and maintainable or do the reverse.
+        """
+        fix_suggestions = """
+        Following specific rules about the usage of blank lines: using them
+        only when necessary.
+        """
+
+
+@BlankLine.subaspect
+class BlankLineAfterDeclaration:
+    """
+    Those found after declarations.
+    """
+    class docs:
+        example = """
+        int main ()
+        {
+          int a;
+          return 0;
+        }
+        """
```

Comment:
This is a bad example.  The last declaration is exempt from the "blank line after declaration rule" as it would interfere with the "trailing blank line" rule.